x-pack/legacy/plugins/index_management/public/components/delete_templates_modal.tsx

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { EuiConfirmModal, EuiOverlayMask, EuiCallOut, EuiCheckbox } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+import React, { Fragment, useState } from 'react';
+import { toastNotifications } from 'ui/notify';
+import { deleteTemplates } from '../services/api';
+import { Template } from '../../common/types';
+
+export const DeleteTemplatesModal = ({
+  templatesToDelete,
+  callback,
+}: {
+  templatesToDelete: Array<Template['name']>;
+  callback: (data?: { hasDeletedTemplates: boolean }) => void;
+}) => {
+  const numTemplatesToDelete = templatesToDelete.length;
+
+  const hasSystemTemplate = Boolean(
+    templatesToDelete.find(templateName => templateName.startsWith('.'))
+  );
+
+  const handleDeleteTemplates = async () => {
+    deleteTemplates(templatesToDelete).then(({ data: { templatesDeleted, errors }, error }) => {
```

Comment:
I haven't looked at the API yet, but it is a bit confusing to have both "errors" and "error". It does seem though that is is a pattern we have used in another app though (now I am the reader and not the writer of the code 😊 ). If `error` (singular) is the Http request that failed, could we rename it "apiError"?